S/satsuma/build_tarballs.jl

```diff
@@ -0,0 +1,60 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "satsuma"
+version = v"0.0.0"
```

Comment:
I don't think the JLL package will be able to be registered with an all-zero version number. It looks like Satsuma has versions (https://github.com/markusa4/satsuma/releases); many build recipes align the JLL version with that of the software they're building.

Relatedly, based on the hash in `GitSource` below, it appears this builds from a commit on `main` rather than building the most recent release. Is that intentional?